pkg/controller/controlplane/genericactuator/actuator.go

```diff
@@ -422,6 +432,9 @@ func (a *actuator) deleteControlPlane(
 		if err := extensionscontroller.DeleteManagedResource(ctx, a.client, cp.Namespace, shootWebhooksResourceName); err != nil {
```

Comment:
Regarding your PR description: Don't you think that it's also necessary to delete the this ManagedResource before deleting the control plane resources?